spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
@viniciusccarvalho I think we're on the same page on the Boot and non-Boot question.

The issue that opened this whole discussion, however, is still unanswered: Boot users are expected to get the project ID from `GcpProperties` but not `GoogleCredentials`, even though `GcpProperties` would contain a `credential` member.

To tackle that a little bit, I renamed the member to `credentialsLocation`, so that it's a little bit clearer to the users that they can't use that member to feed into the adapters (if the `Resource`/`GoogleCredentials` type mismatch wasn't already strong a hint enough :) ). It sounds there isn't much more to do there.